libraries/AP_HAL_Linux/RCInput_Aero.cpp

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (C) 2017  Intel Corporation. All rights reserved.
+ *
+ * This file is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+/* feature added by Anemos Technologies : Philippe Crochat (pcrochat@anemos-technologies.com) */
+
+#include "RCInput_Aero.h"
+#include <stdio.h>
+
+#include <utility>
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_HAL/utility/sparse-endian.h>
+
+using namespace Linux;
+
+#define DEBUG 1
+
+#if DEBUG
+#define debug(fmt, args...) ::printf(fmt "\n", ##args)
+#else
+#define debug(fmt, args...)
+#endif
+
+static const AP_HAL::HAL &hal = AP_HAL::get_HAL();
+
+
+RCInput_Aero::RCInput_Aero()
+{
+	
+}
+
+void RCInput_Aero::init()
+{
+	const uint8_t tx_buffer[2]={CLOCK_REG, 0x00};
+	uint8_t rx_buffer[2];
+	
+	_spi = hal.spi->get_device(DEVICE_NAME);
+	
+	if (!_spi) {
+        AP_HAL::panic("Could not initialize Aero RCInput");
+    }
+	
+    if (!_spi->transfer_fullduplex(tx_buffer, rx_buffer, 2)) {
```

Comment:
you shouldn't actually need tx_buffer and rx_buffer. See how _spi->transfer() is used in other places... Just use different buffers passing the length to each one.

Reply:
I'll have a look

Reply:
it seems that _spi->transfer() does use a buffer in the same way as transfer_fullduplex. Why is that problematic?

Reply:
because you are not doing a fullduplex transfer. You are doing the common "send register address, read value back".

Reply:
got you, but then the intel spi_xfer utility should be changed as well.